src/providers/aws.rs

```diff
@@ -193,8 +189,9 @@ impl<P> AWSLauncher<P> {
     ///
     /// The lifetime of such instances must be declared in advance (1-6 hours). By default, we use 6 hours (the
     /// maximum). If `t` > 6 hours, `AWSLauncher` will use a duration of 6 hours.
-    pub fn set_max_instance_duration(&mut self, t: std::time::Duration) -> &mut Self {
-        self.max_instance_duration = t;
+    pub fn set_max_instance_duration(&mut self, t: i64) -> &mut Self {
```

Comment:
This should probably dictate what happens if you set `t = 0`.

Reply:
yup, forgot to add a call to `min`

Reply:
`max`, not `min`